apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/index.php

```diff
@@ -560,3 +554,92 @@ function wpcom_is_previewing_global_styles( ?int $user_id = null ) {
 	// phpcs:ignore WordPress.Security.NonceVerification.Recommended
 	return ! isset( $_GET['hide-global-styles'] ) && user_can( $user_id, 'administrator' );
 }
+
+/**
+ * Checks whether the site has access to Global Styles with a Personal plan as part of an A/B test.
+ *
+ * @param  int $blog_id Blog ID.
+ * @return bool Whether the site has access to Global Styles with a Personal plan.
+ */
+function wpcom_site_has_global_styles_in_personal_plan( $blog_id = 0 ) {
+	/*
+	 * Users who were assigned to the treatment group of the Global Styles on Personal experiment
+	 * while having a Personal plan should always have access to Global Styles, as long as they still
+	 * have a Personal plan.
+	 */
+	$has_global_styles_in_personal_plan = wpcom_global_styles_has_blog_sticker( 'wpcom-global-styles-personal-plan', $blog_id );
+	if ( $has_global_styles_in_personal_plan ) {
+		$personal_plans = array_filter(
+			wpcom_get_site_purchases( $blog_id ),
+			function ( $purchase ) {
+				return strpos( $purchase->product_slug, 'personal-bundle' ) === 0;
+			}
+		);
+
+		return ! empty( $personal_plans );
+	}
+
+	$owner_id = wpcom_get_blog_owner( $blog_id );
```

Comment:
Should we check whether `function_exists( 'wpcom_get_blog_owner' )`?

Reply:
Not really, this only runs on simple sites.